tests/nlu/extractors/test_lookup_entity_extractor.py

```diff
@@ -0,0 +1,186 @@
+                    "end": 35,
+                    "extractor": "LookupEntityExtractor",
+                },
+                {
+                    "entity": "person",
+                    "value": "Sophie",
+                    "start": 0,
+                    "end": 6,
+                    "extractor": "LookupEntityExtractor",
+                },
+            ],
+        ),
+        (
+            "Rasa is great.",
+            {
+                "city": ["Berlin", "Amsterdam", "New York", "London"],
+                "person": ["Max", "John", "Sophie", "Lisa"],
+            },
+            [],
+        ),
+    ],
+)
+def test_process(
+    text: Text, lookup: Dict[Text, List[Text]], expected_entities: List[Dict[Text, Any]]
+):
+    message = Message(text)
+
+    lookup_table = {}
+    for entity, examples in lookup.items():
+        file_path = write_tmp_file(examples)
+        lookup_table[entity] = file_path
+
+    entity_extractor = LookupEntityExtractor({"lookup": lookup_table})
+    entity_extractor.process(message)
+
+    entities = message.get(ENTITIES)
+    assert entities == expected_entities
+
+
+@pytest.mark.parametrize(
+    "text, lowercase, lookup, expected_entities",
+    [
+        (
+            "berlin and London are cities.",
+            False,
+            {"city": ["Berlin", "Amsterdam", "New York", "London"]},
+            [
+                {
+                    "entity": "city",
+                    "value": "London",
+                    "start": 11,
+                    "end": 17,
+                    "extractor": "LookupEntityExtractor",
+                }
+            ],
+        ),
+        (
+            "berlin and London are cities.",
+            True,
+            {"city": ["Berlin", "Amsterdam", "New York", "london"]},
+            [
+                {
+                    "entity": "city",
+                    "value": "berlin",
+                    "start": 0,
+                    "end": 6,
+                    "extractor": "LookupEntityExtractor",
+                },
+                {
+                    "entity": "city",
+                    "value": "London",
+                    "start": 11,
+                    "end": 17,
+                    "extractor": "LookupEntityExtractor",
+                },
+            ],
+        ),
+    ],
+)
+def test_lowercase(
+    text: Text,
+    lowercase: bool,
+    lookup: Dict[Text, List[Text]],
+    expected_entities: List[Dict[Text, Any]],
+):
+    message = Message(text)
+
+    lookup_table = {}
+    for entity, examples in lookup.items():
+        file_path = write_tmp_file(examples)
+        lookup_table[entity] = file_path
+
+    entity_extractor = LookupEntityExtractor(
+        {"lookup": lookup_table, "lowercase": lowercase}
+    )
+    entity_extractor.process(message)
+
+    entities = message.get(ENTITIES)
+    assert entities == expected_entities
+
+
+def test_do_not_overwrite_any_entities():
+    message = Message("Max lives in Berlin.")
```

Comment:
The lookup table might also still be generating features now that I think of it ... meaning it may not just be spaCy but potentially also the CRF entity detector that picks up similar entities.

Reply:
We don't overwrite any entities, we just add them to the list of already extracted entities. It might happen that if you have multiple entity extractors in the pipeline, the same entity is extracted multiple times. But this should not cause any issue.